sdktests/server_side_events_custom.go

```diff
@@ -18,11 +19,10 @@ func doServerSideCustomEventTests(t *ldtest.T) {
 	t.Run("data and metricValue parameters", doServerSideParameterizedCustomEventTests)
 
 	t.Run("basic properties", func(t *ldtest.T) {
-		users := NewUserFactory("doServerSideCustomEventTests-")
 		metricValue := 1.0
-		for _, anonymousUser := range []bool{false, true} {
-			t.Run(selectString(anonymousUser, "anonymous user", "non-anonymous user"), func(t *ldtest.T) {
-				user := users.NextUniqueUserMaybeAnonymous(anonymousUser)
+		for _, contexts := range data.NewContextFactoriesForAnonymousAndNonAnonymous("doServerSideCustomEventTests") {
+			t.Run(contexts.Description(), func(t *ldtest.T) {
+				user := contexts.NextUniqueContext()
```

Comment:
Here's the deal with the anonymous/non-anonymous stuff. In the old event model, many events had an optional `contextKind` property which was absent for normal users, and set to `"anonymousUser"` for anonymous users— regardless of what other properties the user might have. So, whenever we were testing one of those events, we wanted to make sure its properties are right for each of those two scenarios.

However, it was a bit clunky to keep writing this very specific logic over and over... and, in the new U2C model, those aren't the test cases we want anyway(*), and there won't necessarily be just two of them. So I felt it was better to generalize the logic so that we just call a generator that returns some ContextFactory instances, each of which has a descriptive name and knows how to create the right kind of contexts, and the test just iterates through those.

(* With U2C, there's no `contextKind` property, instead there is `contextKeys` and the variant behaviors we're interested in have to do with single-kind versus multi-kind— plus, there is some other event behavior (key deduplication) that is slightly different for the "user" kind and other single kinds, so there are really three variants of the behavior.)